hoomd/hpmc/pair/pair.py

```diff
@@ -47,12 +47,19 @@ def _attach_hook(self):
         if isinstance(device, hoomd.device.GPU):
             raise RuntimeError("Not implemented on the GPU")
 
-        cpp_sys_def = self._simulation.state._cpp_sys_def
-        cls = getattr(hoomd.hpmc._hpmc, self._cpp_class_name)
-        self._cpp_obj = cls(cpp_sys_def)
+        # Default cpp obj instantiation. Subclasses can override by setting
+        # _cpp_obj before calling super()._attach_hook().
+        if getattr(self, "_cpp_obj", None) is None:
```

Comment:
```suggestion
        if not self._attached:
```
I think this is what you want.

Reply:
I wasn't sure. This is in `_attach_hook` after all. I thought that the object would not be considered attached until `_attach_hook` returns.

Here are I am attempting to avoid code duplication by providing the most common `_cpp_obj` creation protocol, which subclasses can override when needed.

Reply:
I refactored this with a `_make_cpp_obj` method that can be overridden.

Reply:
`_attached` is just do I have a `_cpp_obj` essentially, though this can be overwritten by a subclass as necessary.